hs_access_control/models/user.py

```diff
@@ -1422,12 +1422,15 @@ def can_view_resource(self, this_resource):
         if __debug__:  # during testing only, check argument types and preconditions
             assert isinstance(this_resource, BaseResource)
 
+        access_resource = this_resource.raccess
+
+        if access_resource.public or access_resource.allow_private_sharing:
+            return True
+
         if not self.user.is_active:
             raise PermissionDenied("Requesting user is not active")
 
-        access_resource = this_resource.raccess
-
-        if access_resource.public or self.user.is_superuser:
+        if self.user.is_superuser:
             return True
```

Comment:
better to combine 1433-1434 lines with 1427-1428s by just combining two conditional checks.

Reply:
Can't combine the checks as you suggested as we have to check if the user is active before we can check if the user is super user.

Reply:
OK. Just realized a superuser could be inactive which is why you have to check if a user is inactive first before checking superuser status.